experimental/oak_baremetal_channel/src/server.rs

```diff
@@ -16,15 +16,15 @@
 
 use crate::{message, InvocationChannel, Read, Vec, Write};
 
-pub struct ServerChannelHandle<T: Read + Write> {
-    inner: InvocationChannel<T>,
+pub struct ServerChannelHandle<'a, T: Read + Write + ?Sized> {
+    inner: InvocationChannel<'a, T>,
 }
 
-impl<T> ServerChannelHandle<T>
+impl<'a, T> ServerChannelHandle<'a, T>
 where
-    T: Read + Write,
+    T: Read + Write + ?Sized,
 {
-    pub fn new(socket: T) -> Self {
+    pub fn new(socket: &'a mut T) -> Self {
```

Comment:
I think conceptually it'd be nice for this struct to own the bytestream. We explicitly don't want any other reads/write of bytestream, since it risks making the struct out of sync wrt to the start & end points of messages.

Reply:
Shouldn't the fact that we have a mutable borrow enforce that? You need a `&mut` to be able to call Read or Write, and Rust guarantees there can only be one mutable reference at a time. Therefore, you won't be able to pass one socket to multiple channels.

(But I may be wrong, please double-check my thinking here.)